apps/molgenis-components/src/client/client.ts

```diff
@@ -3,109 +3,63 @@ import { ISchemaMetaData } from "../Interfaces/IMetaData";
 import { IRow } from "../Interfaces/IRow";
 import { ISetting } from "../Interfaces/ISetting";
 import { ITableMetaData } from "../Interfaces/ITableMetaData";
-import {
-  convertRowToPrimaryKey,
-  convertToPascalCase,
-  deepClone,
-} from "../components/utils";
+import { convertToPascalCase, deepClone } from "../components/utils";
 import { IClient, INewClient } from "./IClient";
 import { IQueryMetaData } from "./IQueryMetaData";
 import { columnNames } from "./queryBuilder";
+import { IColumn } from "../Interfaces/IColumn";
 
 // application wide cache for schema meta data
 const schemaCache = new Map<string, ISchemaMetaData>();
+let hitCount = 0;
+let missCount = 0;
 
 export { request };
 const client: IClient = {
   newClient: (schemaName?: string, externalAxios?: Axios): INewClient => {
     const myAxios = externalAxios || axios;
-    // use closure to have metaData cache private to client
-    let schemaMetaData: ISchemaMetaData | null | void = null;
-    let schemaNameCache: string = schemaName || "";
 
     return {
       insertDataRow,
       updateDataRow,
       deleteRow: async (rowKey: IRow, tableName: string) => {
-        return deleteRow(rowKey, tableName, schemaNameCache);
+        return deleteRow(myAxios, rowKey, tableName, schemaName);
       },
       deleteAllTableData: async (tableName: string) => {
-        return deleteAllTableData(tableName, schemaNameCache);
+        return deleteAllTableData(tableName, schemaName);
       },
       fetchSchemaMetaData: async (useCache: boolean = true) => {
```

Comment:
do we have use cases for not wanting to use cache? Would it not make more sense to only add a 'clearCache' function and then for the rest get rid of all the useCache logic?

Reply:
hmm yea , it depends on the context , but for meta data you are probably right ,  if at some point we extend it to data/query cache it could be nice to have it as a option for each method, but that could also be solved using a key ( see nuxt cached fetch fucntions)